File: DebugProject/DebugProjectObjC/DebugProjectObjC/ViewController.m
//
//  ViewController.m
//  DebugProjectObjC
//
//  Created by Tobias Schultka on 30.01.18.
//  Copyright © 2018 GD Mobile GmbH. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

AnnecyMediaSDK* annecyMediaSDK;
NSString* TOKEN = @"6ce0bbf0-2dc8-4d7c-a497-e93105188ba1";
NSString* USER_ID = @"foo";

- (void)viewDidLoad {
    [super viewDidLoad];
    
    annecyMediaSDK = AnnecyMediaSDK.instance;
    annecyMediaSDK.delegate = self;
    [annecyMediaSDK setOptionsWithToken:TOKEN userId:USER_ID];
    [annecyMediaSDK requestOfferwall];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)annecyOnOfferwallReadyWithViewController:(AMSDKViewController * _Nonnull)viewController {
    [self presentViewController:viewController animated:YES completion:nil];
}

@end

File: DebugProject/DebugProjectSwift/DebugProjectSwift/ViewController.swift
//
//  ViewController.swift
//  DebugProjectSwift
//
//  Created by Tobias Schultka on 30.01.18.
//  Copyright © 2018 GD Mobile GmbH. All rights reserved.
//

import UIKit
import AnnecyMediaSDK

class ViewController: UIViewController, AMSDKDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        AnnecyMediaSDK.instance.delegate = self
        AnnecyMediaSDK.instance.options.token = "6ce0bbf0-2dc8-4d7c-a497-e93105188ba1"
        AnnecyMediaSDK.instance.options.userId = "foo"
        
        // Optinal options
        AnnecyMediaSDK.instance.options.buttonBackgroundColor = "FCD"
        AnnecyMediaSDK.instance.options.buttonTextColor = "000"
        AnnecyMediaSDK.instance.options.isHeaderEnabled = true
        AnnecyMediaSDK.instance.options.headline = "Your Offerwall"
        AnnecyMediaSDK.instance.options.headerBackgroundColor = "DCF"
        AnnecyMediaSDK.instance.options.headerTextColor = "333"
        AnnecyMediaSDK.instance.options.headerButtonBackgroundColor = "98C"
        AnnecyMediaSDK.instance.options.headerButtonTextColor = "FFF"
        
        AnnecyMediaSDK.instance.requestOfferwall()
    }
    
    func annecyOnOfferwallReady(viewController: AMSDKViewController) {
        self.present(viewController, animated: true, completion: nil)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
}
